### PR TITLE
Types Added

### LINKED ISSUE #3: Add Typescript types.
Typescript type definitions are currently not exported. An example of where they should be and what this looks like is in the cameraroll repo [here](https://github.com/react-native-community/react-native-picker/blob/5e6cec3e433e710cf80c82ca252bfd04f699514a/.circleci/config.yml#L23-L25).

### PR DESCRIPTION
**Pull request by chakrihacker**
# Summary
Closes #3 

## Test Plan
Let the CI pass

### CODE REVIEW

**Comment by ceo-epiclions** [comment]
Please merge this PR..

**Comment by simonasdev** [comment]
very much so

**Comment by chakrihacker** [comment]
@jainkuniya Can you check over weekend??

**Comment by chakrihacker** [comment]
I took some inspiration from react Native tab view and other community packages.

File: typings/PickerIOS.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { ItemValue } from "./Picker"
+
+export interface PickerIOSItemProps {
+	label?: string;
+	value?: number | string;
+	color?: string;
+	testID?: string;
+}
+
+class PickerIOSItem extends React.Component<PickerIOSItemProps, {}> {}
```
Comment (Naturalclar) [inline typings/PickerIOS.d.ts]: Reflecting changes from https://github.com/DefinitelyTyped/DefinitelyTyped/pull/41502

This line causes TypeScript users to be able to import `PickerItem` which isn't the component's name and may lead to confusion.
```suggestion
```

File: typings/PickerIOS.d.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	itemStyle?: StyleProp<TextStyle>;
+	style?: StyleProp<TextStyle>;
+	onChange?: React.SyntheticEvent<{itemValue: ItemValue, itemIndex: number}>;
+	onValueChange?: (itemValue: ItemValue, itemIndex: number) => void;
+	selectedValue?: ItemValue;
+	testID?: string;
+}
+
+declare class PickerIOS extends React.Component<PickerIOSProps, {}> {
+	static Item: PickerIOSItem
```
Comment (Naturalclar) [inline typings/PickerIOS.d.ts]: Reflecting changes from https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42024

```suggestion
	static Item: React.ComponentType<PickerIOSItem>
```

File: typings/Picker.d.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+   */
+	prompt?: string;
+	/**
+   * Used to locate this view in end-to-end tests.
+   */
+  testID?: string;
+}
+
+declare class Picker extends React.Component<PickerProps, {}> {
+   static Item: PickerItem
```
Comment (Naturalclar) [inline typings/Picker.d.ts]: Reflecting changes from https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42024

```suggestion
	static Item: React.ComponentType<PickerItemProps>
```

File: typings/Picker.d.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+type ItemValue  = number | string
+
+export interface PickerItemProps {
+	label?: string;
+	value: ItemValue;
+	color?: string;
+	testID?: string;
+}
+
+class PickerItem extends React.Component<PickerItemProps, {}> {}
```
Comment (Naturalclar) [inline typings/Picker.d.ts]: Reflecting changes from DefinitelyTyped/DefinitelyTyped#41502

This line causes TypeScript users to be able to import PickerItem which isn't the component's name and may lead to confusion.

```suggestion
```

**Comment by Naturalclar** [comment]
@chakrihacker I've done some tests locally and found some errors with `Picker.Item` component types.
the type definition on `DefinitelyTyped` has been updated recently to fix those changes, so I've made some comments to reflect those changes in this PR as well ;)

**Comment by chakrihacker** [comment]
@Naturalclar thanks, I will update them

File: typings/Picker.d.ts (hunk 3 above)
Comment (chakrihacker) [reply typings/Picker.d.ts]: Changed

File: typings/Picker.d.ts (hunk 4 above)
Comment (chakrihacker) [reply typings/Picker.d.ts]: done

File: typings/Picker.d.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+   */
+	prompt?: string;
+	/**
+   * Used to locate this view in end-to-end tests.
+   */
+  testID?: string;
+}
+
+declare class Picker extends React.Component<PickerProps, {}> {
+   static Item: React.Component<PickerItemProps>;
```
Comment (Naturalclar) [inline typings/Picker.d.ts]: This still causes error when I test this branch locally.
React.ComponentType (defined in `index.d.ts` for React in DefinitelyTyped) would be the type for Class Components.

Also, there are two other static values defined in Picker, which is also defined in `@types/react-native`. could you add them as well?
Below is the code for `@types/react-native`

```
export class Picker extends React.Component<PickerProps> {
    /**
     * On Android, display the options in a dialog.
     */
    static MODE_DIALOG: string;

    /**
     * On Android, display the options in a dropdown (this is the default).
     */
    static MODE_DROPDOWN: string;

    static Item: React.ComponentType<PickerItemProps>;
}
```

```suggestion
   static Item: React.ComponentType<PickerItemProps>;
```

**Comment by Naturalclar** [comment]
LGTM! testing it locally worked fine for me.
@jainkuniya could you take a look at the PR? It would be nice to get some typescript support for this package 🤗

**Comment by jainkuniya** [comment]
Merges it as https://github.com/react-native-community/react-native-picker/commit/cfb9b2d9c743b5a33edc45f138ed1c9e0c48ce41

Thanks again @chakrihacker! for working on this 🎉 .
Thanks @Naturalclar! for reviewing 🎉.

**Pull request closed without merging**